provider/clusterrosaclassic/cluster_rosa_classic_resource.go

```diff
@@ -995,6 +1027,27 @@ func (r *ClusterRosaClassicResource) Read(ctx context.Context, request resource.
 	response.Diagnostics.Append(diags...)
 }
 
+func validateAdminCredentialsChange(ctx context.Context, state,
+	plan *ClusterRosaClassicState) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	if state.AdminCredentials.IsNull() {
+		if common.BoolWithFalseDefault(plan.CreateAdminUser) {
```

Comment:
Nit:
I suggest to create equals function, something like:
```
func adminCredentialsEqual(ctx context.Context, object1 types.Object, object2 types.Object, diags diag.Diagnostics) boolean
```
Then use it for the validation:
```
if ! adminCredentialsEqual(ctx, state.AdminCredentials , plan.AdminCredentials , diags) {
    diags.AddError(common.AssertionErrorSummaryMessage, fmt.Sprintf(common.AssertionErrorDetailsMessage, "admin_credentials", state.AdminCredentials, plan.AdminCredentials))
}
```

Reply:
Or in second thought, why not keep using the deepEqual, like it done before?
```
if !reflect.DeepEqual(state.AdminCredentials, plan.AdminCredentials) {
	diags.AddError(common.AssertionErrorSummaryMessage, fmt.Sprintf(common.AssertionErrorDetailsMessage, "admin_credentials", *state.AdminCredentials, *plan.AdminCredentials))
}
```

Reply:
I prefer to add equals function, because the `AdminCredentials` object has computed value. Even the user tries to apply the same plan as before, the `DeepEqual` will return false when compare `AdminCredentials` in state and plan.
e.g. user provides an empty `AdminCredentials` obj, and after apply, in the state the obj will have the default username and generated password. If user applies the original plan again, the `DeepEqual` func will find the empty `AdminCredentials` obj in plan doesn't equal to the one in state which already have the computed value.
I am not 100% sure about this is the correct behavior in a terraform provider for computed attribute.  What's your thoughts on this? @nirarg

Reply:
Ok, I'm ok with it

Reply:
completed the code refactoring, add the equal func for adminCredential